### PR TITLE
Added analytics support to content extension

### LINKED ISSUE #9841: Rich Notifications : Analytics for Content Extension
We need to add tracking for rich notifications.

Both the service & content extension access / entry should be tracked. When this is combined with the notification type, we can infer the relative engagement as measured by user transitions from the Short Look to the Long Look notification view.

Moreover, we should also ensure that we're tracking user interaction with custom notification actions.

### PR DESCRIPTION
**Pull request by stevebaranski**
## Description
Fixes #9841, which calls for analytics events to track rich notifications engagement. To start, we'll track the launch of the content extension, which is triggered by a user force-touching on the _Short Look_ to evince the _Long Look_ for comment notifications.

## To test

- Verify that the branch builds & existing tests pass.
- While debugging the content extension, manually push a payload with a valid `note_id` value. Instructions for this can be found here: #9924 
- In the _Tracks Live View_, validate that events of type `wpios_notification_content_extension_launched` can be found.

### CODE REVIEW

File: WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// Characterizes the types of service extension events we're interested in tracking.
```
Comment (frosty) [inline WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift]: `service` -> `content`

File: WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+/// The raw value corresponds to the event name in Tracks.
+///
+/// - launched: the service extension was successfully entered & launched
```
Comment (frosty) [inline WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift]: `service` -> `content`

File: WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+///
+private enum ContentExtensionEvents: String {
+    case launched = "wpios_notification_content_extension_launched"
+}
+
+// MARK: - Supports tracking notification service extension events.
```
Comment (frosty) [inline WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift]: `service` -> `content`

File: WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+extension Tracks {
+    /// Tracks the successful launch of the notification service extension.
```
Comment (frosty) [inline WordPress/WordPressNotificationContentExtension/Sources/Tracks/Tracks+ContentExtension.swift]: `service` -> `content`

**Comment by stevebaranski** [comment]
@frosty thank you for the valuable feedback. I have pushed two commits that address the concerns you noted: one for documentation & one to capture the username for both the service & content extensions.

Would you mind taking another look?

**Comment by stevebaranski** [comment]
@frosty thanks for calling out these two additional issues! I have addressed them both via individual commits, and will proceed to merge them once build checks pass.

**Comment by stevebaranski** [comment]
Hound has been reviewing two minor changes to my PR for at least 4 hours. I am going to proceed to merge this. In the unlikely event that Hound should report an issue, I'll submit a new PR.

**Pull request merged**